From 450da8164fdb2dda83b2c441c56a918e997661db Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 17 Apr 2024 15:03:33 +0200
Subject: [PATCH 584/841] feat(stm32mp1): remove stop watchdog when going in
 low power

Low power phases are only executed after the initialization of OP-TEE.
Remove low power configuration of watchdog in TF-A.
The configuration to stop watchdog during standby and s2idle is done in
OP-TEE.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 8315c3c9b28a9b4e0cd19d131a7f938af77825c7 feat(stm32mp1): remove stop watchdog when going in low power]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/366906
Change-Id: I7d8fea47b5610127c9c1243fe6c471849296e661
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/372968
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/iwdg/stm32_iwdg.c            |  6 ----
 plat/st/common/include/stm32mp_common.h |  5 ---
 plat/st/stm32mp1/stm32mp1_private.c     | 45 -------------------------
 3 files changed, 56 deletions(-)

diff --git a/drivers/st/iwdg/stm32_iwdg.c b/drivers/st/iwdg/stm32_iwdg.c
index 6d08cceb5..9fd09504f 100644
--- a/drivers/st/iwdg/stm32_iwdg.c
+++ b/drivers/st/iwdg/stm32_iwdg.c
@@ -159,12 +159,6 @@ int stm32_iwdg_init(void)
 			stm32mp_register_secure_periph_iomem(iwdg->base);
 		}
 
-#if defined(IMAGE_BL2)
-		if (stm32_iwdg_shadow_update(idx, iwdg->flags) != BSEC_OK) {
-			return -1;
-		}
-#endif
-
 		if ((hw_init & IWDG_HW_ENABLED) == 0) {
 			/* Use default timeout, ignore DT's "timeout-sec" */
 			stm32_iwdg_start(iwdg);
diff --git a/plat/st/common/include/stm32mp_common.h b/plat/st/common/include/stm32mp_common.h
index 2c983ada9..beb113d32 100644
--- a/plat/st/common/include/stm32mp_common.h
+++ b/plat/st/common/include/stm32mp_common.h
@@ -75,11 +75,6 @@ uint32_t stm32_iwdg_get_instance(uintptr_t base);
 /* Return bitflag mask for expected IWDG configuration from OTP content */
 uint32_t stm32_iwdg_get_otp_config(uint32_t iwdg_inst);
 
-#if defined(IMAGE_BL2)
-/* Update OTP shadow registers with IWDG configuration from device tree */
-uint32_t stm32_iwdg_shadow_update(uint32_t iwdg_inst, uint32_t flags);
-#endif
-
 #if STM32MP_UART_PROGRAMMER || !defined(IMAGE_BL2)
 /* Get the UART address from its instance number */
 uintptr_t get_uart_address(uint32_t instance_nb);
diff --git a/plat/st/stm32mp1/stm32mp1_private.c b/plat/st/stm32mp1/stm32mp1_private.c
index e9a0a1bb8..77e1644ca 100644
--- a/plat/st/stm32mp1/stm32mp1_private.c
+++ b/plat/st/stm32mp1/stm32mp1_private.c
@@ -605,54 +605,9 @@ uint32_t stm32_iwdg_get_otp_config(uint32_t iwdg_inst)
 		iwdg_cfg |= IWDG_HW_ENABLED;
 	}
 
-	if ((otp_value & BIT(iwdg_inst + HW2_OTP_IWDG_FZ_STOP_POS)) != 0U) {
-		iwdg_cfg |= IWDG_DISABLE_ON_STOP;
-	}
-
-	if ((otp_value & BIT(iwdg_inst + HW2_OTP_IWDG_FZ_STANDBY_POS)) != 0U) {
-		iwdg_cfg |= IWDG_DISABLE_ON_STANDBY;
-	}
-
 	return iwdg_cfg;
 }
 
-#if defined(IMAGE_BL2)
-uint32_t stm32_iwdg_shadow_update(uint32_t iwdg_inst, uint32_t flags)
-{
-	uint32_t otp_value;
-	uint32_t otp;
-	uint32_t result;
-
-	if (stm32_get_otp_index(HW2_OTP, &otp, NULL) != 0) {
-		panic();
-	}
-
-	if (stm32_get_otp_value(HW2_OTP, &otp_value) != 0) {
-		panic();
-	}
-
-	if ((flags & IWDG_DISABLE_ON_STOP) != 0) {
-		otp_value |= BIT(iwdg_inst + HW2_OTP_IWDG_FZ_STOP_POS);
-	}
-
-	if ((flags & IWDG_DISABLE_ON_STANDBY) != 0) {
-		otp_value |= BIT(iwdg_inst + HW2_OTP_IWDG_FZ_STANDBY_POS);
-	}
-
-	result = bsec_write_otp(otp_value, otp);
-	if (result != BSEC_OK) {
-		return result;
-	}
-
-	/* Sticky lock OTP_IWDG (read and write) */
-	if ((bsec_set_sr_lock(otp) != BSEC_OK) ||
-	    (bsec_set_sw_lock(otp) != BSEC_OK)) {
-		return BSEC_LOCK_FAIL;
-	}
-
-	return BSEC_OK;
-}
-#endif
 
 bool stm32mp1_addr_inside_backupsram(uintptr_t addr)
 {
-- 
2.47.2

